### PR TITLE
Modifying schema validation messages

### LINKED ISSUE #181: Provide better Error Validation messages.
Improve the clarity of error validation messages.

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #181 

Before:

screen shot 2019-02-25 at 12 34 22 pm

After:

selection_016

---

Before:

screen shot 2019-02-25 at 12 22 16 pm

After:

selection_019

--- 

Before:
screen shot 2019-02-25 at 12 08 45 pm

After: (Uses single quotes now)

selection_021

---

Before:

screen shot 2019-02-25 at 12 24 39 pm

After:

selection_022

---

Before:

screen shot 2019-02-25 at 12 36 37 pm

After:

selection_024

---

Before:

screen shot 2019-02-25 at 12 17 24 pm

After:

selection_026

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+# "Identity constraints". And we should cover all of the following errors.
+
+#validation (3.X.4)
+
+        cvc-attribute.3 = cvc-attribute.3: The value ''{2}'' of attribute ''{1}'' on element ''{0}'' is not valid with respect to its type, ''{3}''.
+        cvc-attribute.4 = cvc-attribute.4: The value ''{2}'' of attribute ''{1}'' on element ''{0}'' is not valid with respect to its fixed '{'value constraint'}'. The attribute must have a value of ''{3}''.
+        cvc-complex-type.2.1 = cvc-complex-type.2.1: Element ''{0}'' must have no character or element information item [children], because the type''s content type is empty.
+        cvc-complex-type.2.2 = cvc-complex-type.2.2: Element ''{0}'' must have no element [children], and the value must be valid.
+    # changed    
+				cvc-complex-type.2.3 = Element, \"{0}\", cannot contain text content.\nThe content type is defined as element-only.\n\nCode:
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties]: keep single quotes instead of doubles

File: org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+		# changed
+				cvc-complex-type.2.4.a = Invalid element name:\n\t\"{0}\"\nExpects one of the following:\n{1}\nError indicated by:\n\t{2}\nwith code:
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties]: single quotes

File: org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    # changed    
+				cvc-complex-type.2.4.b = Child elements are missing from element:\n\t"{0}\"\n\nThe following elements are expected:\n{1}\nError indicated by\n{2}\nwith code:
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties]: single quotes

File: org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        cvc-complex-type.3.1 = cvc-complex-type.3.1: Value ''{2}'' of attribute ''{1}'' of element ''{0}'' is not valid with respect to the corresponding attribute use. Attribute ''{1}'' has a fixed value of ''{3}''.
+        cvc-complex-type.3.2.1 = cvc-complex-type.3.2.1: Element ''{0}'' does not have an attribute wildcard for attribute ''{1}''.
+        cvc-complex-type.3.2.2 = cvc-complex-type.3.2.2: Attribute ''{1}'' is not allowed to appear in element ''{0}''.
+    # changed    
+				cvc-complex-type.4 = Attribute:\n\t''{1}''\nis required in element:\n\t''{0}''\n\nCode:
+        cvc-complex-type.5.1 = cvc-complex-type.5.1: In element ''{0}'', attribute ''{1}'' is a Wild ID. But there is already a Wild ID ''{2}''. There can be only one.
+        cvc-complex-type.5.2 = cvc-complex-type.5.2: In element ''{0}'', attribute ''{1}'' is a Wild ID. But there is already an attribute ''{2}'' derived from ID among the '{'attribute uses'}'.
+    # changed    
+				cvc-datatype-valid.1.2.1 = Content of type \"{1}\" is expected.\n\nThe following is not a valid type:\n\t\"{0}\".\n\nCode:
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties]: single quotes

File: org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+        cvc-elt.3.2.2 = cvc-elt.3.2.2: There must be no fixed '{'value constraint'}' for element ''{0}'', because ''{1}'' is specified.
+        cvc-elt.4.1 = cvc-elt.4.1: The value ''{2}'' of attribute ''{1}'' of element ''{0}'' is not a valid QName.
+        cvc-elt.4.2 = cvc-elt.4.2: Cannot resolve ''{1}'' to a type definition for element ''{0}''.
+        cvc-elt.4.3 = cvc-elt.4.3: Type ''{1}'' is not validly derived from the type definition, ''{2}'', of element ''{0}''.
+        cvc-elt.5.1.1 = cvc-elt.5.1.1: '{'value constraint'}' ''{2}'' of element ''{0}'' is not a valid default value for type ''{1}''.
+        cvc-elt.5.2.2.1 = cvc-elt.5.2.2.1: Element ''{0}'' must have no element information item [children].
+        cvc-elt.5.2.2.2.1 = cvc-elt.5.2.2.2.1: The value ''{1}'' of element ''{0}'' does not match the fixed '{'value constraint'}' value ''{2}''.
+        cvc-elt.5.2.2.2.2 = cvc-elt.5.2.2.2.2: The value ''{1}'' of element ''{0}'' does not match the '{'value constraint'}' value ''{2}''.
+    # changed    
+				cvc-enumeration-valid = Value \"{0}\" is not in the enumeration list.\n\nIt must be one of the following:\n{1}\nCode:
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/resources/XMLSchemaMessages.properties]: single quotes

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+		MultiLineStream stream = new MultiLineStream(names, 0);
+
+		while (!stream.eos()) { // }
+			stream.advance(1);// Consume ' ' or '{' if first item
+			if(hasNamespace) {
+				stream.advance(1); // Consume "
+				stream.advanceUntilAnyOfChars(_DQO, _SQO, _SIQ); // " | " | '
+				stream.advance(2); // Consume quotation and':'
+			}
+			sb.append("\t\"");
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java]: sb.append(" - ");

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+			while (stream.peekChar() != _CCB && stream.peekChar() != _CMA) { // } | ,
+				sb.append(Character.toString((char) stream.peekChar()));
+				stream.advance(1);
+			}
+			sb.append("\"\n");
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java]: remove double quote

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+	 * @return
+	 */
+	private static String reformatArrayElementNames(String names) {
+		StringBuilder sb = new StringBuilder();
+
+		MultiLineStream stream = new MultiLineStream(names, 0);
+
+		while (!stream.eos()) { // ]
+			stream.advance(1);// Consume ' ' or '[' if first item
+			sb.append("\t\"");
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java]: sb.append(" - ");

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+			while (stream.peekChar() != _CSB && stream.peekChar() != _CMA) { // ] | ,
+				sb.append(Character.toString((char) stream.peekChar()));
+				stream.advance(1);
+			}
+			sb.append("\"\n");
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/LSPMessageFormatter.java]: remove double quote

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLFileAssociationsDiagnosticsTest.java [hunk 10]
```diff
@@ -41,7 +41,7 @@
  * XML file associations diagnostics tests.
  */
 public class XMLFileAssociationsDiagnosticsTest {
-
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLFileAssociationsDiagnosticsTest.java]: unnecessary change

**Comment by NikolasKomonen** [comment]
@fbricon updated with example images

**Comment by NikolasKomonen** [comment]
@fbricon 

 * Updated with comparison images
 * The case with multiple missing attributes has multiple error messages (no changes needed)
 * Consistent with single quotations
